### PR TITLE
Database Integration and Mentor Screen

### PR DESCRIPTION
**Pull request by IcemanOmikron**
Logging in now contacts the database to verify info. Basic mentor screen is just copy of student screen.

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -28,6 +31,16 @@ class MyHomePage extends StatefulWidget {
 class _MyHomePageState extends State<MyHomePage> {
   TextStyle style = TextStyle(fontFamily: 'Montserrat', fontSize: 20.0);
 
+  final _emailTextController = TextEditingController();
+  final _passwordTextController = TextEditingController();
+
+  @override
+  void dispose(){
```
Comment (chwerth) [inline lib/main.dart:38]: Is this implicitly called at some point? I don't see any calls to it

**Review by chwerth — APPROVED** [review]
Seems good to me. We should probably switch to named routes at some point still.

File: lib/main.dart (hunk 1 above)
Comment (IcemanOmikron) [reply lib/main.dart:38]: Yes, dispose is implicitly called when the widget is disposed (i.e. going to new screen) but TextEditingControllers are not automatically disposed when their parent widget is disposed. So it is necessary to override the widget's dispose to also dispose of the controllers.

Comment (chwerth) [reply lib/main.dart:38]: Nice, I didn't know that.

**Review by ggpepp — APPROVED** [review]

**Pull request merged**